aiida_cp2k/utils/parser.py

```diff
@@ -43,6 +43,235 @@ def parse_cp2k_output(fobj):
     return result_dict
 
 
+def parse_cp2k_output_advanced(fobj):  # pylint: disable=too-many-locals, too-many-statements, too-many-branches
```

Comment:
Can this function be splitted into smaller sub-parts? Also, could you please add a couple of pytests for that?

Reply:
I don't think I can, since they are a list of `if` instances on the line which is read. I can take out some parts but this would make the readability much more complicated. I propose to keep it  like this since it is still quite experimental.